File: Trabajo pasado/src/icc/ajedrez/Pieza.java
package icc.ajedrez;
import java.util.List;

/**Clase abstracta padre de las piezas de ajedrez.
*@author Sebastián A.
*/
public abstract class Pieza extends Posicion {

	private boolean color;

	/**Método constructor para las piezas.
	*@param columna Columna en la que irá la pieza.
	*@param renglón Renglón en el que irá la pieza.
	*@param color Color de la pieza.
	*/
	public Pieza(char columna,int renglón, boolean color) {
		super(columna, renglón);
		this.color = color;
	}

	/**Método para saber todos los posibles movimientos de la pieza.
	*@return Lista con los posibles movimientos de la pieza en la forma (columna,renglón).
	*/
	public abstract List posiblesMovimientos();

	/**Método getter para obtener el color de la pieza.
	*@return Color de la pieza.
	*/
	public boolean getColor() {
		return color;
	}

	/**Método para verificar si una pieza puede realizar cierto movimiento.
	*@param columna Columna de la posición que se quiere saber si es posible.
	*@param renglón Renglón de la posición que se quiere saber si es posible.
	*@return Devuelve "true" si la pieza puede llegar a dicho lugar y "false" en caso contrario.
	*/
	public boolean esVálida(char columna, int renglón) {

		List posiblesMovs = posiblesMovimientos();
		String posición = "("+columna+","+renglón+")";

		if ( posiblesMovs.contains(posición) ) {
			return true;
		}
		return false;

	} //Fin de esVálida().

	/**Método para obtener la pieza representada en forma de String.
	*@return Representación en cadena de la pieza.
	*/
	public String toString() {
		return "Esta pieza se encuentra en la posición: ("+getColumna()+","+getRenglón()+").";
	}

} //Fin de clase Pieza.

File: code/Pieza.rb
require_relative './Posicion'

# Clase que representa una pieza del ajedrez.
class Pieza

  # Método getter para la posición.
  attr_reader :posición

  # Método getter para el color.
  attr_reader :color

  # Inicializador de una pieza. La crea e indica
  # en qué lugar se encuentra y de qué color es.
  # columna:: Columna en la que se encontrará la pieza.
  # renglón:: Renglón en el que se encontrará la pieza.
  # color:: Color que tendrá la pieza (booleano).
  def initialize(columna, renglón, color)
    @posición = Posicion.new(columna, renglón)
    @color = color
  end

  # Devuelve una lista con todos los posibles
  # movimientos de la pieza.
  def posiblesMovimientos
    raise "Método sin implementación; las clases " +
          "hijas deben implementarlo."
  end

  # Devuelve la posición de la pieza desplazada "x" lugares
  # horizontalmente y "y" lugares verticalmente.
  # x:: Número de columnas por desplazar.
  # y:: Número de renglones por desplazar.
  def desplazar(x, y)
    a = @posición.columna + x
    b = @posición.renglón + y
    return Posicion.new(a, b)
  end

  # Método to_s de las piezas.
  def to_s
    # Color de la pieza...
    c = @color ? "blanca" : "negra"

    # Construcción del string...
    # NOTA: como cada pieza (Caballo, Reina, Rey, etc.) llamará a este
    # método por su cuenta, self.class no devuelve "Pieza" si no la
    # clase de correspondiente a la pieza que llame el método...
    s = "Esta pieza #{c} (#{self.class}) se encuentra en "
    s += "la posición #{@posición}."
    return s
  end

end